Webapp/sources/blueprints/save_reaction/routes.py

```diff
@@ -441,13 +489,59 @@ def autosave_sketcher() -> Response:
 
     current_time = datetime.now(pytz.timezone("Europe/London")).replace(tzinfo=None)
     reaction_smiles = str(request.form["reactionSmiles"])
+    reaction_rxn = str(request.form["reactionRXN"])
+
+    update_dict = {
+        "time_of_update": current_time,
+        "reaction_smiles": reaction_smiles,
+        "reaction_rxn": reaction_rxn,
+    }
+    reaction.update(**update_dict)
+    feedback = "Reaction Updated!"
+    return jsonify({"feedback": feedback})
+
+
+@save_reaction_bp.route("/_save_polymer_mode", methods=["POST"])
+@login_required
+def save_polymer_mode():
+    """Updates reaction dict with polymer mode"""
+    reaction = services.reaction.get_current_from_request()
+    services.auth.edit_reaction(reaction)
 
-    update_dict = {"time_of_update": current_time, "reaction_smiles": reaction_smiles}
+    polymer_mode = request.form["polymerMode"]
+    polymer_mode = polymer_mode.lower() == "true"  # convert string to boolean
+    update_dict = {"polymer_mode": polymer_mode}
     reaction.update(**update_dict)
     feedback = "Reaction Updated!"
     return jsonify({"feedback": feedback})
 
 
+@save_reaction_bp.route("/_get_polymer_mode", methods=["GET"])
+@login_required
+def get_polymer_mode():
+    """Read reaction dict to get polymer mode"""
+    reaction_id = str(request.args.get("reactionID"))
+    workgroup_name = str(request.args.get("workgroup"))
+    workbook_name = str(request.args.get("workbook"))
+    workbook = services.workbook.get_workbook_from_group_book_name_combination(
+        workgroup_name, workbook_name
+    )
+    abort_if_user_not_in_workbook(workgroup_name, workbook_name, workbook)
+    reaction = (
```

Comment:
should be able to replace the query with this function in services https://github.com/AI4Green/AI4Green/blob/8e6295c9d78e440ddfefe82aa0a5710f074e20fc/Webapp/sources/services/reaction.py#L50